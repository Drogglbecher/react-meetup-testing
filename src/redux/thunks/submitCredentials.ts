// Redux
import {Dispatch} from 'redux';

// Actions
import {
  Action,
  SubmitCredentialsStartAction,
  failed,
  submitCredentialsStart,
  submitCredentialsSuccess,
} from '../actions/authenticationActions';

export const submitCredentials = (username: string, password: string) => (
  dispatch: Dispatch<Action | SubmitCredentialsStartAction>,
) => {
  dispatch(submitCredentialsStart());

  if (!username || !password) {
    return dispatch(failed('Invalid authentication data.'));
  } else if (password === username) {
    return dispatch(submitCredentialsSuccess(username));
  } else {
    return dispatch(failed('Username and password are not the same.'));
  }
};
