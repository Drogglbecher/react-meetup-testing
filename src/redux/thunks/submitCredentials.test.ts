// Actions
import {submitCredentials} from './submitCredentials';

describe('submitCredentials', () => {
  const dispatch = jest.fn();

  it('should dispatch `[AUTHENTICATION] Failed` when `submitCredentials` is called with invalid username or password', () => {
    const state = {
      username: '',
      password: 'password',
    };

    submitCredentials(state.username, state.password)(dispatch);

    expect(dispatch.mock.calls).toMatchSnapshot();
  });

  it('should dispatch `[AUTHENTICATION] Failed` when `submitCredentials` is called with different username and password', () => {
    const state = {
      username: 'user',
      password: 'password',
    };

    submitCredentials(state.username, state.password)(dispatch);

    expect(dispatch.mock.calls).toMatchSnapshot();
  });

  it('should dispatch `[AUTHENTICATION] SubmitCredentialsSuccess` when `submitHandler` is called with same username and password', () => {
    const state = {
      username: 'user',
      password: 'user',
    };

    submitCredentials(state.username, state.password)(dispatch);

    expect(dispatch.mock.calls).toMatchSnapshot();
  });
});
