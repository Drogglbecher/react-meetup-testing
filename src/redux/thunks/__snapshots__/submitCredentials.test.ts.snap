// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitCredentials should dispatch \`[AUTHENTICATION] Failed\` when \`submitCredentials\` is called with different username and password 1`] = `
Array [
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "Invalid authentication data.",
      "type": "[AUTHENTICATION] Failed",
    },
  ],
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "Username and password are not the same.",
      "type": "[AUTHENTICATION] Failed",
    },
  ],
]
`;

exports[`submitCredentials should dispatch \`[AUTHENTICATION] Failed\` when \`submitCredentials\` is called with invalid username or password 1`] = `
Array [
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "Invalid authentication data.",
      "type": "[AUTHENTICATION] Failed",
    },
  ],
]
`;

exports[`submitCredentials should dispatch \`[AUTHENTICATION] SubmitCredentialsSuccess\` when \`submitHandler\` is called with same username and password 1`] = `
Array [
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "Invalid authentication data.",
      "type": "[AUTHENTICATION] Failed",
    },
  ],
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "Username and password are not the same.",
      "type": "[AUTHENTICATION] Failed",
    },
  ],
  Array [
    Object {
      "type": "[AUTHENTICATION] SubmitCredentialsStart",
    },
  ],
  Array [
    Object {
      "payload": "user",
      "type": "[AUTHENTICATION] SubmitCredentialsSuccess",
    },
  ],
]
`;
